### PR TITLE
initial implementation

### PR DESCRIPTION
**Pull request by morethanreal**
When working on `paper-input` I wanted an input that notifies with the `input` event as well as setting the `value` via script, so I made this. PTAL @cdata @notwaldorf

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "iron-input",
+  "version": "0.8.0",
+  "authors": "The Polymer Authors",
```
Comment (notwaldorf) [inline bower.json]: nit: i think this is an array in `seed-element`

File: index.html [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<!doctype html>
+<!--
+  @license
```
Comment (notwaldorf) [inline index.html]: nit: i think this shouldn't be indented (maybe)

**Comment by notwaldorf** [comment]
Cool! :+1:

File: demo/index.html [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    </section>
+
+  </template>
+
+  <script>
+
+    var scope = document.querySelector('template[is=x-autobind]');
+
+    scope.setValue = function(event) {
```
Comment (cdata) [inline demo/index.html]: Just curious: was this actually working for you with `rc.6`? I @blasten was trying this yesterday and it didn't seem to work (poking x-autobind with a handler).

File: test/iron-input.html [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    <template>
+      <input is="iron-input" bind-value="foobar">
+    </template>
+  </test-fixture>
+
+  <script>
+
+    suite('basic', function() {
+
+      test('setting bindValue sets value', function() {
```
Comment (cdata) [inline test/iron-input.html]: Woot, thanks for the specs :beers:

File: iron-input.html [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+<link rel="import" href="../polymer/polymer.html">
+
+<!--
+By default you can only get notified of changes to an `input`'s `value` due to user input:
+
+    <input value="{{myValue::input}}">
+
+`iron-input` adds the `bind-value` property that mirrors the `value` property, and can be used
+for two-way data binding. `bind-value` will notify if it is changed either by user input or by script.
```
Comment (cdata) [inline iron-input.html]: It's annoying that this is necessary :crying_cat_face:

**Comment by cdata** [comment]
LGTM :+1:

File: demo/index.html (hunk 3 above)
Comment (morethanreal) [inline demo/index.html]: Yeah, it works for me?

**Pull request merged**